### ISSUE TITLE
(5P) External Media: Focus loss a11y

### ISSUE THREAD

**Issue by obenland** [Automattic/jetpack#15828]
There is a focus loss when leaving the core Media Image Modal with the Select Image button in place. I haven't traced the bug down yet, but I think we're overriding the default implementation on how to access the core media image modal, so we may need to build a way to handle bringing the focus back after the modal is closed. The Google Photos and Pexel ones work more-or-less correctly. Ideally focus is just returned from where it came.

**Comment by jeryj** [comment]
I've confirmed that this is not an issue in Gutenberg. In Gutenberg, the Image block "Media Library" button opens the media library modal. Closing it returns focus to the Media Library button. 

I'm guessing the behavior is still the same. It's looking for a Media Library button to return the focus to, but that button is now within a closed menu.

**Comment by jeryj** [comment]
One more thing to note: On an already uploaded image, there's a Replace button to open a dropdown, which has an Open Media Library button within it. When the modal is closed, it returns focus to the opened dropdown (Open Media Library button, in this case). The implementation they use is different than ours. I wonder if this is part of the issue: https://github.com/WordPress/gutenberg/blob/master/packages/block-editor/src/components/media-replace-flow/index.js#L126

Do we need to restructure this to have `<MediaUpload>` render the Media Library button?

For the Pexels and Google Photos buttons, we should likely follow the same pattern and return focus to the opened dropdown item that was clicked.

**Comment by jeryj** [comment]
> Do we need to restructure this to have `<MediaUpload>` render the Media Library button?

This doesn't seem to be the case, but I really can't track down why it works in other places (like the Replace flow) and not in the Image block. I'm still thinking the best pattern should be to return focus to the open dropdown menu item though.

**Comment by WunderBart** [comment]
Fixed by https://github.com/Automattic/jetpack/pull/16102
